File: 70-climbing-stairs/climbing-stairs.java
class Solution {
    public int climbStairs(int n) {
        if (n <= 3) {
            return n;
        }
        int f = 1;
        int s = 2;
        int t = f + s;
        n -= 2;
        while(n > 0) {
            t = f + s;
            f = s;
            s = t;
            n--;
        }
        return t;
    }
}

File: 70-climbing-stairs/climbing-stairs.rs
impl Solution {
    pub fn climb_stairs(n: i32) -> i32 {
        let sqrt5 = 5.0_f64.sqrt();
        let phi = (1.0 + sqrt5) / 2.0;
        let psi = (1.0 - sqrt5) / 2.0;
        ((phi.powi(n + 1) - psi.powi(n + 1)) / sqrt5).round() as i32
    }
}